### PR TITLE
fix tx receipt JSON types; permit more null event fields

### PR DESCRIPTION
**Pull request by jmank88**
This PR relaxes `FilterEvent` parsing to accept more null fields, and fixes a type issue in `TransactionReceipt.fromMap`.

### CODE REVIEW

File: lib/src/core/transaction_information.dart [hunk 1]
```diff
@@ -148,8 +148,8 @@ class TransactionReceipt {
             ? (hexToDartInt(map['status'] as String) == 1)
             : null,
         logs = map['logs'] != null
-            ? (map['logs'] as List<Map<String, dynamic>>)
-                .map((log) => FilterEvent.fromMap(log))
+            ? (map['logs'] as List<dynamic>)
```
Comment (simolus3) [inline lib/src/core/transaction_information.dart:151]: Maybe we should serialize and then de-serialize the input map in `transaction_information_test.dart`. E.g.

```dart
test('parses full object', ()  {
  final rawMap = {
    'blockHash': '...',
    // ...
  }
  // ensure we can parse nested structures without type information
  final parsed = TransactionReceipt.fromMap(json.decode(json.encode(rawMap)) as Map<String, dynamic>);
})
```
These errors are easy to overlook, and I think that would catch this error.

Comment (jmank88) [reply lib/src/core/transaction_information.dart:151]: What about just starting with the raw JSON?

File: test/core/transaction_information_test.dart [hunk 2]
```diff
@@ -1,44 +1,26 @@
...
+        '"transactionIndex":"0x0","address":"0x16c5785ac562ff41e2dcfdf829c5a142f1fccd7d",'
+        '"data":"0x0000000000000000000000000000000000000000000000000000000000000000",'
+        '"topics":["0x59ebeb90bc63057b6515673c3ecf9438e5058bca0f92585014eced636878c9a5"]}],'
+        '"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",'
+        '"root":"0x89628cd74b7246a144781e0f537bac145df645945c213f82ab45f4c6729f1e4c",'
+        '"to":"0xf8c59caf9bb8a7a2991160b592ac123108d88f7b",'
+        '"transactionHash":"0xb75a96c4751ff03b1bdcf5300e80a45e788e52650b0a4e2294e7496c215f4c9d",'
+        '"transactionIndex":"0x18","status":"0x1"}') as Map<String, dynamic>);
```
Comment (simolus3) [inline test/core/transaction_information_test.dart:23]: This works too, but I'd probably use a triple string literal for easier reading:
```dart
final parsed = TransactionReceipt.fromMap(json.decode('''
{
  "blockHash": "0x5548b5f215b99674c7f23c9a701a005b5c18e4a963b55163eddada54562ac521",
  "blockNumber": "0x18",
  "contractAddress": "0x6671e02bb8bd3a234b13d79d1c285a9df657233d",
  "cumulativeGasUsed": "0x4cc5f",
  "from": "0xf8c59caf9bb8a7a2991160b592ac123108d88f7b",
  "gasUsed": "0x4cc5f",
  "logs": [
    {
      "logIndex": "0x1",
      "blockNumber": "0x14",
      "blockHash": "0x8216c5785ac562ff41e2dcfdf5785ac562ff41e2dcfdf829c5a142f1fccd7d",
      "transactionHash": "0xdf829c5a142f1fccd7d8216c5785ac562ff41e2dcfdf5785ac562ff41e2dcf",
      "transactionIndex": "0x0",
      "address": "0x16c5785ac562ff41e2dcfdf829c5a142f1fccd7d",
       "data": "0x0000000000000000000000000000000000000000000000000000000000000000",
       "topics": ["0x59ebeb90bc63057b6515673c3ecf9438e5058bca0f92585014eced636878c9a5"]
    }
  ],
  "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "root": "0x89628cd74b7246a144781e0f537bac145df645945c213f82ab45f4c6729f1e4c",
  "to": "0xf8c59caf9bb8a7a2991160b592ac123108d88f7b",
  "transactionHash": "0xb75a96c4751ff03b1bdcf5300e80a45e788e52650b0a4e2294e7496c215f4c9d",
  "transactionIndex": "0x18",
  "status": "0x1"
}
  ''') as Map<String, dynamic>);
```

And sorry for the tedious labor that probably went into splitting the string literals like that.

Comment (jmank88) [reply test/core/transaction_information_test.dart]: Ah much better! nbd, the IDE most of the work

**Pull request merged**